cmd/influxd/server_integration_test.go

```diff
@@ -619,6 +619,44 @@ func runTestsData(t *testing.T, testName string, nodes Cluster, database, retent
 		},
 		{
 			reset: true,
+			name: "first value",
+			write: `{"database" : "%DB%", "retentionPolicy" : "%RP%", "points": [
+				{"name": "cpu", "timestamp": "2000-01-01T00:00:00Z", "fields": {"value": 2}},
+				{"name": "cpu", "timestamp": "2000-01-01T00:01:00Z", "fields": {"value": 7}},
+				{"name": "cpu", "timestamp": "2000-01-01T00:01:10Z", "fields": {"value": 9}}
+			]}`,
+			query: `SELECT first(value) FROM cpu`,
+			queryDb:  "%DB%",
+			// FIXME: returned time should be "2000-01-01T00:00:00Z"
+			expected: `{"results":[{"series":[{"name":"cpu","columns":["time","first"],"values":[["1970-01-01T00:00:00Z",2]]}]}]}`,
+		},
+		{
+			reset: true,
+			name: "last value",
+			write: `{"database" : "%DB%", "retentionPolicy" : "%RP%", "points": [
+				{"name": "cpu", "timestamp": "2000-01-01T00:00:00Z", "fields": {"value": 2}},
+				{"name": "cpu", "timestamp": "2000-01-01T00:01:00Z", "fields": {"value": 7}},
+				{"name": "cpu", "timestamp": "2000-01-01T00:01:10Z", "fields": {"value": 9}}
+			]}`,
+			query: `SELECT last(value) FROM cpu`,
+			queryDb:  "%DB%",
+			// FIXME: returned time should be "2000-01-01T00:01:10Z"
+			expected: `{"results":[{"series":[{"name":"cpu","columns":["time","last"],"values":[["1970-01-01T00:00:00Z",9]]}]}]}`,
+		},
+		{
+			reset: true,
```

Comment:
It's by purpose. I think it's better to make every test to stand on its own in case of any future modifications.

> 在 2015年5月1日，06:49，otoolep <notifications@github.com> 写道：
> 
> In cmd/influxd/server_integration_test.go:
> 
> > +		},
> > +		{
> > +			reset: true,
> > +			name: "last value",
> > +			write: `{"database" : "%DB%", "retentionPolicy" : "%RP%", "points": [
> > +				{"name": "cpu", "timestamp": "2000-01-01T00:00:00Z", "fields": {"value": 2}},
> > +				{"name": "cpu", "timestamp": "2000-01-01T00:01:00Z", "fields": {"value": 7}},
> > +				{"name": "cpu", "timestamp": "2000-01-01T00:01:10Z", "fields": {"value": 9}}
> > +			]}`,
> > +			query: `SELECT last(value) FROM cpu`,
> > +			queryDb:  "%DB%",
> > +			// FIXME: returned time should be "2000-01-01T00:01:10Z"
> > +			expected: `{"results":[{"series":[{"name":"cpu","columns":["time","last"],"values":[["1970-01-01T00:00:00Z",9]]}]}]}`,
> > +		},
> > +		{
> > +			reset: true,
> The data is the same with every write, so there is no need to set the reset flag between each test. It will just slow testing down without reason.
> 
> —
> Reply to this email directly or view it on GitHub.
>